### ISSUE TITLE
Unify tryrun scripts

### ISSUE THREAD

**Issue by MichalStrehovsky** [dotnet/runtime#86829]
https://github.com/dotnet/runtime/pull/86781#discussion_r1207000325

We have tryrun.cmake that is mostly used for `-cross` builds.

Then we have similar things in [runtime/src/native/libs/configure.cmake](https://github.com/dotnet/runtime/blob/c5256458235e00c4bc4634c3cc389fefdd4c10d8/src/native/libs/configure.cmake#L572-L605) and https://github.com/dotnet/runtime/blob/main/eng/native/tryrun_ios_tvos.cmake.

We should reconcile this.

Related: #56622

**Comment by janvorli** [comment]
@MichalStrehovsky the `tryrun.cmake` contains values for cross build that the configure.cmake would detect if the build was executed on the target OS. It is used only for checks that execute code, so they cannot be evaluated on the build hosting OS. This is a cmake concept - the `tryrun.cmake` is meant to be separate, as it is loaded by cmake as an initialization of its cache.
It is not clear to me how we could reasonably unify these.

**Comment by MichalStrehovsky** [comment]
> @MichalStrehovsky the `tryrun.cmake` contains values for cross build that the configure.cmake would detect if the build was executed on the target OS. It is used only for checks that execute code, so they cannot be evaluated on the build hosting OS.

Yes, it's the same thing that the quoted part of configure.cmake is also doing. It's not clear why we spread these around the codebase

**Comment by janvorli** [comment]
Oh, I am surprised to see it in the configure.cmake in libraries. The ones in coreclr don't have that, they use the tryrun.cmake.

**Comment by akoeplinger** [comment]
@ViktorHofer it can stay in 8.0 for now

**Comment by ilonatommy** [comment]
@akoeplinger, this looks like a fast cleanup issue, will you have capacity for it?

**Comment by akoeplinger** [comment]
Hopefully 😄
